apl/scalar-functions/ip-functions/parse-ipv4.mdx

```diff
@@ -0,0 +1,143 @@
+---
+title: parse_ipv4
+description: 'This page explains how to use the parse_ipv4 function in APL.'
+---
+
+The `parse_ipv4` function in APL extracts the four octets of an IPv4 address and represents them as integers. You can use this function to break down an IPv4 address into its constituent components for advanced analysis, filtering, or comparisons. It is especially useful for tasks like analyzing network traffic logs, identifying trends in IP address usage, or performing security-related queries.
+
+## For users of other query languages
+
+If you come from other query languages, this section explains how to adjust your existing queries to achieve the same results in APL.
+
+<AccordionGroup>
+<Accordion title="Splunk SPL users">
+
+In Splunk SPL, extracting IPv4 components requires using regular expressions or string manipulation. APL simplifies this process with the dedicated `parse_ipv4` function.
+
+<CodeGroup>
+```sql Splunk example
+| eval octets=split("192.168.1.1", ".") | table octets
+```
+
+```kusto APL equivalent
+['sample-http-logs']
+| extend octets = parse_ipv4('192.168.1.1')
+| project octets
+```
+</CodeGroup>
+
+</Accordion>
+<Accordion title="ANSI SQL users">
+
+In ANSI SQL, breaking down an IPv4 address often requires using functions like `SUBSTRING` or `SPLIT`. APL offers the `parse_ipv4` function as a straightforward alternative.
+
+<CodeGroup>
+```sql SQL example
+SELECT SPLIT_PART(ip, '.', 1) AS octet1, SPLIT_PART(ip, '.', 2) AS octet2 FROM logs
+```
+
+```kusto APL equivalent
+['sample-http-logs']
+| extend octets = parse_ipv4('192.168.1.1')
+| project octets
+```
+</CodeGroup>
+
+</Accordion>
+</AccordionGroup>
+
+## Usage
+
+### Syntax
+
+```kusto
+parse_ipv4(ipv4_address)
+```
+
+### Parameters
+
+| Parameter       | Type   | Description                                    |
+|------------------|--------|------------------------------------------------|
+| `ipv4_address`   | string | The IPv4 address to parse into integer octets. |
+
+### Returns
+
+The function returns an array of four integers, each representing an octet of the IPv4 address.
+
+## Use case examples
+
+<Tabs>
+<Tab title="Log analysis">
+
+You can use the `parse_ipv4` function to analyze web traffic by breaking down user IP addresses into octets.
+
+**Query**
+
+```kusto
+['sample-http-logs']
+| extend ip_octets = parse_ipv4('192.168.1.1')
+| project _time, uri, method, ip_octets
```

Comment:
So I don't know how to make this into a reasonable example. But for a lot of corps they split up the private IP range into sites. So 10.35.* is Texas and 10.36.* is California, etc. They'd probably filter on 10 being the first digit and then group based on the second. Something like that? Not quite sure on how to make it something easy to understand.